pennylane/templates/embeddings/amplitude.py

```diff
@@ -164,51 +168,57 @@ def _preprocess(features, wires, pad_with, normalize):
 
         * If features is batched, the processing that follows is applied to each feature set in the batch.
         * Check that the features tensor is one-dimensional.
-        * If pad_with is None, check that the first dimension of the features tensor
+        * If pad_with is None, check that the last dimension of the features tensor
           has length :math:`2^n` where :math:`n` is the number of qubits. Else check that the
-          first dimension of the features tensor is not larger than :math:`2^n` and pad features
+          last dimension of the features tensor is not larger than :math:`2^n` and pad features
           with value if necessary.
-        * If normalize is false, check that first dimension of features is normalised to one. Else, normalise the
+        * If normalize is false, check that last dimension of features is normalised to one. Else, normalise the
           features tensor.
         """
 
         # check if features is batched
-        batched = len(qml.math.shape(features)) > 1
+        batched = qml.math.ndim(features) > 1
+
+        if batched and qml.math.get_interface(features) == "tensorflow":
+            raise ValueError("AmplitudeEmbedding does not support batched Tensorflow features.")
 
         features_batch = features if batched else [features]
 
+        new_features_batch = []
```

Comment:
Could we mutate `features_batch` with the new solution as before? I.e., not needing a `new_features_batch` list?

Reply:
unfortunately not, because we want to allow broadcasted inputs that are tensors and the `feature_set` changes shape. For example the following breaks:
```pycon
>>> x = np.ones((3, 3))
>>> x[1] = np.ones(4)
ValueError: could not broadcast input array from shape (4) into shape (3)
```